crates/rpc-proto/proto/note.proto

```diff
@@ -6,42 +6,83 @@ import "block.proto";
 import "digest.proto";
 import "merkle.proto";
 
+// Represents a note's metadata.
 message NoteMetadata {
+    // The account which sent the note.
     account.AccountId sender = 1;
+
+    // The type of the note (0b01 = public, 0b10 = private, 0b11 = encrypted).
     uint32 note_type = 2;
+
+    // A value which can be used by the recipient(s) to identify notes intended for them.
     fixed32 tag = 3;
+
+    // Specifies when a note is ready to be consumed:
+    // (6 least significant bits - hint identifier (tag), bits 6 to 38 - Hint payload).
+    // See `miden_objects::notes::execution_hint` for more info.
     fixed64 execution_hint = 4;
+
+    // An arbitrary user-defined value.
     fixed64 aux = 5;
 }
 
+// Represents a note.
 message Note {
+    // The block number in which the note was created.
     fixed32 block_num = 1;
+
+    // The index of the note in the block.
     uint32 note_index = 2;
+
+    // The ID of the note.
     digest.Digest note_id = 3;
+
+    // The note metadata.
     NoteMetadata metadata = 4;
+
+    // The note inclusion proof in the block.
     merkle.MerklePath merkle_path = 5;
+
     // This field will be present when the note is public.
     // details contain the `Note` in a serialized format.
     optional bytes details = 6;
 }
 
+// Represents proof of a note's inclusion in a block.
 message NoteInclusionInBlockProof {
+    // The ID of the note.
     digest.Digest note_id = 1;
```

Comment:
Should we mention that this is a commitment to the underlying note data - i.e., "A unique identifier of the note which is a 32-byte commitment to the underlying note data."